src/details/ArborX_DetailsSortUtils.hpp

```diff
@@ -56,9 +56,9 @@ namespace Details
 {
 
 // NOTE returns the permutation indices **and** sorts the morton codes
-template <typename DeviceType>
+template <typename NumberType, typename DeviceType>
```

Comment:
Is there a reason to not template this function on the `View` itself? In addition, `NumberType` sounds awkward to me (I assume this is common in deal.II), and would prefer `ValueType`.